### DISCUSSION TITLE
new Plugin tested / Exiftool

### DISCUSSION THREAD

**Discussion by ragebear00** [dnGrep/dnGrep#1124] [Show and tell]
Exiftool is a useful tool to read and edit metadata of many files (jpeg, bmp, pdf, tif and etc) you can find more at [https://exiftool.org](https://exiftool.org/)

You may download exiftool.exe save in the Plugin subfolder Exiftool and use the follwing dnGREP.ExifTool.plugin
```
Name=Exiftool
Application=exiftool.exe
Arguments= -a -u -g1 "{inputFile}"
WorkingDirectory=C:\Software\dnGREP\dnGrep\Plugins\ExifTool
Extensions=jpeg, jpg, tif, tiff
```
You can now search all information in the metadata including camera model. geo location, and XMP which is new standard to save long string over 255. 

![image](https://github.com/dnGrep/dnGrep/assets/168278723/9c9d11bc-ce8d-4ebf-afd1-6cc70b6f5292)

**Comment by VladTester255** [comment]
Maybe someone will find it helpful to have a different file name (portable), without entering the path for the portable version of dnGrep and with a different language selection (replace with your language):
```
Name=Exiftool
Application=exiftool(-k).exe
Arguments= -a -u -g1 -lang ru "{inputFile}"
WorkingDirectory=
Extensions=jpeg, jpg, tif, tiff
```
Is there an option to substitute the language depending on the selected language in dnGrep settings?
```
Available languages:
  cs - Czech (─Мe┼бtina)
  de - German (Deutsch)
  en - English
  en-ca - Canadian English
  en-gb - British English
  es - Spanish (Espa├▒ol)
  fi - Finnish (Suomi)
  fr - French (Fran├зais)
  it - Italian (Italiano)
  ja - Japanese (цЧецЬмшкЮ)
  ko - Korean (эХЬъ╡ньЦ┤)
  nl - Dutch (Nederlands)
  pl - Polish (Polski)
  ru - Russian (╨а╤Г╤Б╤Б╨║╨╕╨╣)
  sk - Slovak (Sloven─Нina)
  sv - Swedish (Svenska)
  tr - Turkish (T├╝rk├зe)
  zh-cn - Simplified Chinese (чоАф╜Уф╕нцЦЗ)
  zh-tw - Traditional Chinese (ч╣БщлФф╕нцЦЗ)
No file specified
```

**Comment by doug24** [comment]
The WorkingDirectory is optional and will default to the folder where the .plugin file is located. It is not required, but you can put the application file, jar files, script files and any other files you need in the working directory ([reference](https://github.com/dnGrep/dnGrep/wiki/Bring-Your-Own-Plugin)).  

No, there is no provision for passing the current language selection to a plugin command.  There isn't a one-to-one mapping between the languages in dnGrep translations and the ones you list above. Just set your chosen language in the arguments. It is unlikely to change for a given user.
